src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/VSMSBuildNuGetProject.cs

```diff
@@ -35,17 +34,14 @@ public class VSMSBuildNuGetProject : MSBuildNuGetProject
             InternalMetadata.Add(NuGetProjectMetadataKeys.ProjectId, projectId);
         }
 
-        public override Task<IReadOnlyList<IDependencyGraphProject>> GetDirectProjectReferencesAsync(DependencyGraphCacheContext context)
+        public override async Task<IReadOnlyList<IDependencyGraphProject>> GetDirectProjectReferencesAsync(DependencyGraphCacheContext context)
         {
             IReadOnlyList<IDependencyGraphProject> references = null;
-            if (context != null && context.DirectReferenceCache.TryGetValue(MSBuildProjectPath, out references))
-            {
-
-            }
-            else
+            if (context == null || !context.DirectReferenceCache.TryGetValue(MSBuildProjectPath, out references))
             {
                 var solutionManager = (VSSolutionManager)ServiceLocator.GetInstance<ISolutionManager>();
                 var list = new List<IDependencyGraphProject>();
+                await Microsoft.VisualStudio.Shell.ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
Everything inside the if block here should be inside a JTF so that it doesn't bring the caller to the UI thread also